code/datums/elements/volatile_gas_storage.dm

```diff
@@ -0,0 +1,46 @@
+/datum/element/volatile_gas_storage
+	element_flags = ELEMENT_BESPOKE
+	id_arg_index = 2
+
+	var/minimum_explosive_pressure
+	var/max_explosive_pressure
+	var/max_explosive_force
+
+/datum/element/volatile_gas_storage/Attach(datum/target, minimum_explosive_pressure=5000, max_explosive_pressure=100000, max_explosive_force=9)
+	. = ..()
+	if(istype(target, /obj/machinery/portable_atmospherics))
+		RegisterSignal(target, COMSIG_OBJ_BREAK, .proc/PortableBreak)
+	else if(istype(target, /obj/machinery/atmospherics/components))
+		RegisterSignal(target, COMSIG_OBJ_BREAK, .proc/AtmosComponentBreak)
+	else
+		return ELEMENT_INCOMPATIBLE
+
+	src.minimum_explosive_pressure = minimum_explosive_pressure
+	src.max_explosive_pressure = max_explosive_pressure
+	src.max_explosive_force = max_explosive_force
+
+/datum/element/volatile_gas_storage/Detach(datum/source, ...)
+	. = ..()
+	UnregisterSignal(source, COMSIG_OBJ_BREAK)
+
+/datum/element/volatile_gas_storage/proc/Break(atom/origin, datum/gas_mixture/released_gas)
+	var/expelled_pressure = min(released_gas?.return_pressure(), max_explosive_pressure)
+
+	if(expelled_pressure < minimum_explosive_pressure)
+		return
```

Comment:
Is there a reason you're not using origin.return_air() for this? are you worried about pain related to getting the air of a turf rather then the air inside something?

Reply:
I forgot that proc existed but yes it's the better choice here